astro/src/components/GitHubEdit.astro

```diff
@@ -0,0 +1,49 @@
+---
+import type { AstroGlobal } from "astro";
+
+/** Gets the URL to edit the page on GitHub */
+
+export function getGithubEditUrl(Astro: Readonly<AstroGlobal>) {
+  const slug = Astro.params?.slug;
+  const path = Astro.url.pathname;
+  const splitSlug = slug?.split("/");
+
+  // Each file layout is unique
+  let appendFile = "";
+  if (path.includes("/articles/")) {
+    appendFile = splitSlug?.length
+      ? splitSlug?.length > 1
+        ? path.replace(`${slug}`, "") + `${slug}.md`
+        : path + `index.mdx`
+      : path;
+  } else if (path.includes("/dev-tools/")) {
+    appendFile = splitSlug?.length
+      ? path.replace(/.$/, `.mdx`)
+      : path + `index.mdx`;
+  } else if (path.includes("/quickstarts/")) {
+    appendFile = splitSlug?.length
+      ? path.replace("/docs", "").replace(/.$/, `.mdx`)
+      : path.replace("/docs", "");
+  }
+  return `https://github.com/FusionAuth/fusionauth-site/blob/master/astro/src/content${appendFile}`;
```

Comment:
Can you please add a fallback because we'll be moving more content over and may forget to update this. I think a decent fallback would be to just link to the top level fusionauth-site repo or maybe just to https://github.com/FusionAuth/fusionauth-site/blob/master/astro/src/content

Also how does this work on the index page https://fusionauth.io/docs/quickstarts/ or https://fusionauth.io/articles/

Reply:
The fallback already is in place as it wont match and defaults to `https://github.com/FusionAuth/fusionauth-site/blob/master/astro/src/content`.

`http://localhost:3000/docs/quickstarts/` takes you to `https://github.com/FusionAuth/fusionauth-site/tree/master/astro/src/content/quickstarts`

`http://localhost:3000/articles/` takes you to `https://github.com/FusionAuth/fusionauth-site/tree/master/astro/src/content/articles`

The reason I broke that out is each of those folders had an index as well so something like `http://localhost:3000/articles/authentication/` will take you direct to the index file `https://github.com/FusionAuth/fusionauth-site/blob/master/astro/src/content/articles/authentication/index.mdx`

Reply:
Awesome, I missed that.